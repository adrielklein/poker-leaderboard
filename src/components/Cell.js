import * as React from 'react';
import Input from '@material-ui/core/Input';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import TableCell from '@material-ui/core/TableCell';
import {
  Table, TableEditRow,
} from '@devexpress/dx-react-grid-material-ui';
import { countries } from '../constants.js';

const getColor = (amount) => {
  if (amount < 3000) {
    return '#F44336';
  }
  if (amount < 5000) {
    return '#FFC107';
  }
  if (amount < 8000) {
    return '#FF5722';
  }
  return '#009688';
};

export const HighlightedCell = ({
                                  tableColumn, value, classes, children, style,
                                }) => (
  <TableCell
    style={{
      color: getColor(value),
      textAlign: tableColumn.align,
      ...style,
    }}
  >
    {children}
  </TableCell>
);

export const LookupEditCell = ({
                          availableColumnValues, value, onValueChange,
                        }) => (
  <TableCell
  >
    <Select
      value={value}
      onChange={event => onValueChange(event.target.value)}
      input={(
        <Input
        />
      )}
    >
      {availableColumnValues.map(item => (
        <MenuItem key={item} value={item}>
          {item}
        </MenuItem>
      ))}
    </Select>
  </TableCell>
);

export const Cell = (props) => {
  const { column } = props;
  if (column.name === 'amount') {
    return <HighlightedCell {...props} />;
  }
  return <Table.Cell {...props} />;
};

export const EditCell = (props) => {
  const { column } = props;
  if (column.name === 'country') {
    return <LookupEditCell {...props} availableColumnValues={countries} />;
  }
  return <TableEditRow.Cell {...props} />;
};
