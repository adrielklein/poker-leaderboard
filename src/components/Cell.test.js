import React from 'react';
import { HighlightedCell, LookupEditCell, Cell, EditCell } from './Cell';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

it('HighlightedCell', () => {
  const component = shallow(<HighlightedCell tableColumn={ { align: 'center' } }/>);
  expect(toJson(component)).toMatchSnapshot();
});

it('LookupEditCell', () => {
  const component = shallow(<LookupEditCell />);
  expect(toJson(component)).toMatchSnapshot();
});

it('Cell', () => {
  const component = shallow(<Cell />);
  expect(toJson(component)).toMatchSnapshot();
});

it('EditCell', () => {
  const component = shallow(<EditCell />);
  expect(toJson(component)).toMatchSnapshot();
});
