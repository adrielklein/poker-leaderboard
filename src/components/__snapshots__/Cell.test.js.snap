// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightedCell 1`] = `
<WithStyles(TableCell)
  style={
    Object {
      "color": "#009688",
      "textAlign": "center",
    }
  }
/>
`;
